ultralytics/models/yolo/detect/predict.py

```diff
@@ -22,14 +22,20 @@ class DetectionPredictor(BasePredictor):
 
     def postprocess(self, preds, img, orig_imgs):
         """Post-processes predictions and returns a list of Results objects."""
-        preds = ops.non_max_suppression(
-            preds,
-            self.args.conf,
-            self.args.iou,
-            agnostic=self.args.agnostic_nms,
-            max_det=self.args.max_det,
-            classes=self.args.classes,
-        )
+        if isinstance(preds, (list, tuple)):
+            preds = preds[0]
+
+        if self.end2end:
+            preds = [pred[pred[:, 4] > self.args.conf] for pred in preds]
+        else:
+            preds = ops.non_max_suppression(
+                preds,
+                self.args.conf,
+                self.args.iou,
+                agnostic=self.args.agnostic_nms,
+                max_det=self.args.max_det,
+                classes=self.args.classes,
+            )
```

Comment:
Do we really need to avoid NMS here? It seems like we could avoid a lot of customization here if we just treat all models identically, and the NMS speeds for these low-grid count e2e models will be negligible.

I'm just seeing a ton of self.end2end additions throughout the codebase, which won't scale well if we have hundreds of models to support at some point, each with some "special" attributes.